wacksbywarby/wack.py

```diff
@@ -71,38 +75,41 @@ def await_pizza_party(discord, num_sales):
         discord.send_party_message()
 
 
-def is_sale(current_num_sales):
-    logger.info("Checking if is_sale")
-    previous_num_sales = Wackabase.get_last_entry().get("num_sales", 0)
-    logger.info(f"current num sales: {current_num_sales}, previously stored num sales: {previous_num_sales}")
-    return current_num_sales > previous_num_sales
-
-
-def main(dry=False):
+def main(db, dry=False):
     try:
         logger.info("TIME TO WACK")
-        load_dotenv()
         logger.info("Dry run: %s", dry)
         etsy = Etsy(debug=dry)
         discord = Discord(debug=dry)
-        id_to_listing_diff = etsy.get_inventory_state_diff()
+
+        previous_inventory = db.get_last_entry()
+        current_inventory = etsy.get_inventory_state()
+        id_to_listing_diff = etsy.get_inventory_state_diff(
+            previous_inventory, current_inventory
+        )
```

Comment:
what if werby manually increases the inventory. we would return here but wouldn't update the last_num_sales or the inventory? is it easy to add a test case to test that the DBs are updated in addition to the discord messages?

Reply:
if there's an increase in inventory, the `id_to_listing_diff` will not be falsy, so it shouldn't return here. There is already a test case for this— added a comment for it!